plugins/print/plugin.js

```diff
@@ -6,41 +6,54 @@
 /**
  * @fileOverview Print Plugin
  */
+( function() {
+	'use strict';
 
-CKEDITOR.plugins.add( 'print', {
-	// jscs:disable maximumLineLength
-	lang: 'af,ar,az,bg,bn,bs,ca,cs,cy,da,de,de-ch,el,en,en-au,en-ca,en-gb,eo,es,es-mx,et,eu,fa,fi,fo,fr,fr-ca,gl,gu,he,hi,hr,hu,id,is,it,ja,ka,km,ko,ku,lt,lv,mk,mn,ms,nb,nl,no,oc,pl,pt,pt-br,ro,ru,si,sk,sl,sq,sr,sr-latn,sv,th,tr,tt,ug,uk,vi,zh,zh-cn', // %REMOVE_LINE_CORE%
-	// jscs:enable maximumLineLength
-	icons: 'print,', // %REMOVE_LINE_CORE%
-	hidpi: true, // %REMOVE_LINE_CORE%
-	init: function( editor ) {
-		// Print plugin isn't available in inline mode yet.
-		if ( editor.elementMode == CKEDITOR.ELEMENT_MODE_INLINE )
-			return;
+	CKEDITOR.plugins.add( 'print', {
+		requires: 'preview',
+		// jscs:disable maximumLineLength
+		lang: 'af,ar,az,bg,bn,bs,ca,cs,cy,da,de,de-ch,el,en,en-au,en-ca,en-gb,eo,es,es-mx,et,eu,fa,fi,fo,fr,fr-ca,gl,gu,he,hi,hr,hu,id,is,it,ja,ka,km,ko,ku,lt,lv,mk,mn,ms,nb,nl,no,oc,pl,pt,pt-br,ro,ru,si,sk,sl,sq,sr,sr-latn,sv,th,tr,tt,ug,uk,vi,zh,zh-cn', // %REMOVE_LINE_CORE%
+		// jscs:enable maximumLineLength
+		icons: 'print,', // %REMOVE_LINE_CORE%
+		hidpi: true, // %REMOVE_LINE_CORE%
+		init: function( editor ) {
+			var pluginName = 'print';
 
-		var pluginName = 'print';
+			editor.addCommand( pluginName, CKEDITOR.plugins.print );
+			editor.ui.addButton && editor.ui.addButton( 'Print', {
+				label: editor.lang.print.toolbar,
+				command: pluginName,
+				toolbar: 'document,50'
+			} );
+		}
+	} );
 
-		// Register the command.
-		editor.addCommand( pluginName, CKEDITOR.plugins.print );
+	/**
+	 * Allows to customise implementation of printing the editor's content,
+	 * provided by [Print](https://ckeditor.com/cke4/addon/print) plugin.
+	 * The provided value should be of {@link CKEDITOR.commandDefinition} type.
+	 *
+	 * @singleton
+	 * @class CKEDITOR.plugins.print
+	 */
+	CKEDITOR.plugins.print = {
+		exec: function( editor ) {
+			var previewWindow = CKEDITOR.plugins.preview.createPreview( editor ).$;
 
-		// Register the toolbar button.
-		editor.ui.addButton && editor.ui.addButton( 'Print', {
-			label: editor.lang.print.toolbar,
-			command: pluginName,
-			toolbar: 'document,50'
-		} );
-	}
-} );
+			if ( !previewWindow ) {
+				return;
+			}
 
-CKEDITOR.plugins.print = {
-	exec: function( editor ) {
-		if ( CKEDITOR.env.gecko ) {
-			editor.window.$.print();
-		} else {
-			editor.document.$.execCommand( 'Print' );
-		}
-	},
-	canUndo: false,
-	readOnly: 1,
-	modes: { wysiwyg: 1 }
-};
+			if ( CKEDITOR.env.gecko ) {
+				previewWindow.print();
+			} else {
+				previewWindow.document.execCommand( 'Print' );
+			}
+
+			previewWindow.close();
+		},
+		canUndo: false,
+		readOnly: 1,
+		modes: { wysiwyg: 1 }
```

Comment:
Mode for `print` is more specific, as `preview` seems to work also in `sourcearea` - which actually seems a bit weird as it shows up editor data, no source code. I'm for using the same modes for both plugins, but it seems like `preview` plugin mode should be decreased to `wysiwyg` only, so it doesn't close us for `sourcearea` preview/print feature. WDYT?